### PR TITLE
mark: init at version 3.5

### PR DESCRIPTION
**Pull request by rguevara84**
* A tool for syncing your markdown documentation with Atlassian
  Confluence pages

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/mark/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    owner  = "kovetskiy";
+    repo   = "mark";
+    rev    = "${version}";
+    sha256 = "0za4n2caqr3gflfxr1hdd328g7r52h7x0ws7r9mjvdnmwjgc0b2b";
+  };
+
+  vendorSha256 = "1hvizcg5b3y2qgjiw77rb795xz9w1dzr3b8q2ji48ihll27g9f0m";
+
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/text/mark/default.nix:17]:
```suggestion
  vendorSha256 = "1hvizcg5b3y2qgjiw77rb795xz9w1dzr3b8q2ji48ihll27g9f0m";

  meta = with lib; {
```

File: pkgs/tools/text/mark/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
```
Comment (SuperSandro2000) [inline pkgs/tools/text/mark/default.nix:1]:
```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/mark/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  vendorSha256 = "1hvizcg5b3y2qgjiw77rb795xz9w1dzr3b8q2ji48ihll27g9f0m";
+
+
+  meta = with stdenv.lib; {
+    description = "A tool for syncing your markdown documentation with Atlassian Confluence pages";
+    homepage = "https://github.com/kovetskiy/mark";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ rguevara84 ];
```
Comment (SuperSandro2000) [inline pkgs/tools/text/mark/default.nix:21]: Please add yourself to maintainers.nix according to the contribution guide.

File: pkgs/tools/text/mark/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "mark";
+  version = "3.5";
+
+  src = fetchFromGitHub {
+    owner  = "kovetskiy";
+    repo   = "mark";
+    rev    = "${version}";
```
Comment (kalbasit) [inline pkgs/tools/text/mark/default.nix:10]:
```suggestion
    rev    = version;
```

File: pkgs/tools/text/mark/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub }:
```
Comment (kalbasit) [inline pkgs/tools/text/mark/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

I don't believe it's used.

**Review by kalbasit — COMMENTED** [review]
Looking at their Makefile, there's a way to embed the version in the binary by passing `-X main.version=${version}` as ldflags. I'm afk so I don't have the syntax at hand, you can grep nixpkgs for ldflags in files that also has buildGoModule to find an example; Once you get it done, ask mark to display its version to double check that it worked.

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -7477,6 +7477,12 @@
     githubId = 42433779;
     name = "Rémy Grünblatt";
   };
+  rguevara84 = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7480]: Please put this into a separate commit with the message ``maintainers: add rguevara84``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by kalbasit** [comment]
@rguevara84 can you squash the init with the pr feedback. At the end you should have two commit, one for adding you as a maintainer and the other for mark init.

**Comment by kalbasit** [comment]
@GrahamcOfBorg build mark

**Review by kalbasit — APPROVED** [review]

**Comment by kalbasit** [comment]
```
[nix-shell:~/.cache/nixpkgs-review/pr-110996]$ ./results/mark/bin/mark --version
3.5
```

Looks good.

@SuperSandro2000 PTAL.

**Pull request merged**